conans/test/unittests/client/generators/cmake_find_package_multi_test.py

```diff
@@ -44,12 +44,36 @@ def test_cmake_find_package_multi_links_flags():
     cpp_info = CppInfo("mypkg", "dummy_root_folder1")
     # https://github.com/conan-io/conan/issues/8811 regression, fix with explicit - instead of /
     cpp_info.sharedlinkflags = ["-NODEFAULTLIB", "-OTHERFLAG"]
-    cpp_info.exelinkflags = ["-OPT:NOICF"]
+    cpp_info.exelinkflags = ["-OPT:NOICF", "-s FULL_ES3=1"]
     conanfile.deps_cpp_info.add("mypkg", cpp_info)
 
     gen = CMakeFindPackageMultiGenerator(conanfile)
     files = gen.content
     d = files["mypkgTarget-release.cmake"]
-    assert "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:-OPT:NOICF>" in d
-    assert "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:-NODEFAULTLIB;-OTHERFLAG>" in d
-    assert "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:-NODEFAULTLIB;-OTHERFLAG>" in d
+    assert '"$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:-OPT:NOICF;-s FULL_ES3=1>"' in d
+    assert '"$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:-NODEFAULTLIB;-OTHERFLAG>"' in d
+    assert '"$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:-NODEFAULTLIB;-OTHERFLAG>"' in d
+
+
+def test_components_multi_links_flags():
+    conanfile = ConanFile(Mock(), None)
+    conanfile.settings = "os", "compiler", "build_type", "arch"
+    conanfile.initialize(Settings({"os": ["Windows"],
+                                   "compiler": ["gcc"],
+                                   "build_type": ["Release"],
+                                   "arch": ["x86"]}), EnvValues())
+    conanfile.settings.build_type = "Release"
+    conanfile.settings.arch = "x86"
+
+    cpp_info = CppInfo("mypkg", "dummy_root_folder1")
+    cpp_info.components["cmp1"].sharedlinkflags = ["-NODEFAULTLIB", "-OTHERFLAG"]
+    cpp_info.components["cmp1"].exelinkflags = ["-OPT:NOICF", "-s FULL_ES3=1"]
+    conanfile.deps_cpp_info.add("mypkg", DepCppInfo(cpp_info))
+
+    gen = CMakeFindPackageMultiGenerator(conanfile)
+    files = gen.content
+    d = files["mypkgTarget-release.cmake"]
+
+    assert '"$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:-OPT:NOICF;-s FULL_ES3=1>"' in d
+    assert '"$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:-NODEFAULTLIB;-OTHERFLAG>"' in d
+    assert '"$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:-NODEFAULTLIB;-OTHERFLAG>"' in d
```

Comment:
This test is fine, but I think it could be better to have a real one. We could add it in `conans/test/functional/generators/cmake_find_package_multi_test.py` and check that it's correctly running with some real link flags like the PR link pasted in the previous comment.